include/libhal/servo/rc.hpp

```diff
@@ -44,36 +52,70 @@ class rc_servo : public hal::servo
       "The maximum microseconds is greater than the period of the frequency");
     constexpr auto frequency = hertz{ Frequency };
 
+    // Check if any errors happened while setting the frequency of the pwm.
+    // Using HAL_CHECK will cascade any errors that happen out of the factory
+    // function.
```

Comment:
```suggestion
    // Using HAL_CHECK will return any errors that occur from the factory
    // function allowing the caller to choose what to do with the error information.
```

Something more like that?